pkg/controller/controller.go

```diff
@@ -1267,6 +1289,113 @@ func (c *M3DBController) getParentCluster(pod *corev1.Pod) (*myspec.M3DBCluster,
 	return cluster, nil
 }
 
+// cleanupUnusedStorageForSts removes all PVCs and PVs for the stateful which are not used.
+// This method should be called only when the cluster is stable
+// and there are no ongoing deployments.
```

Comment:
term `deployment` is probably foreign to operator domain. Operator simply tries to reach certain cluster state based on specification we set on M3DB crd. I think stable here means stateful sets match CRD definition and all pods are ready.